### PR TITLE
Clean up the pop gen model appendix

### PR DESCRIPTION
**Pull request by apragsdale**
This is in an attempt to iron out the flow of the population genetics section in the appendix, remove comments and sections that have been resolved, and add some detail where needed. I really edited this section heavily, and plan to do a bit more. Comments and suggestions would be great!

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -476,137 +476,156 @@ \section*{Appendix}
...
+terms are interchangeable.
 A population is defined as some collection of individuals that exists for
 some period of time, and has a well-defined size (i.e., number of individuals)
 during that time period. Individuals can move between populations
-either through ancestor-descendant relationships between the populations,
-or through processes of continuous or pulse migration.
-No other properties of the populations are specified in the model:
-we are concerned only with defining the populations, their sizes, and the
+either according to their ancestor-descendant relationships
+or through processes of continuous or pulse migrations.
```
Comment (molpopgen) [inline paper.tex]: maybe simpler to say "processes involving migration"?

File: paper.tex [hunk 2]
```diff
@@ -476,137 +476,156 @@ \section*{Appendix}
...
+of the $i^{th}$ allele currently at frequency $p_i$
+one generation in the next generation is $p_iw_i/\bar{w}$,
+where $w_i$ and $\bar{w}$ are the marginal and
+mean fitnesses, respectively, properly weighted according to ancestry proportions.
+In this framework,
+a forward-time simulation of finite populations is equivalent
+to multinomial sampling of allele frequencies each
+generation (\citet[][pp 29-31]{burger2000-ul}, \citet[][pp 179-181]{crowkimura1970}),
+and a backwards-time (coalescent) simulation follows the approximations
+described in \cite[][chapter 3]{wakeley2008-hd}.
```
Comment (molpopgen) [inline paper.tex]: Wakeley is a fine ref here, but Hudson and especially Tajima are probably worth including.  Also, and practically, it may be harder for future readers to get a hold of Wakeley's book.

File: paper.tex [hunk 3]
```diff
@@ -476,137 +476,156 @@ \section*{Appendix}
...
+Further, this model assumes "soft" selection \citep{christiansen1975hard},
```
Comment (molpopgen) [inline paper.tex]: LaTeX  quotes here, ``soft''. (That was my mistake!)

File: paper.tex [hunk 4]
```diff
@@ -476,137 +476,156 @@ \section*{Appendix}
...
+can sum to more than one.
+Each generation a proportion of offspring
+$\sigma$ are expected to be generated through clonal reproduction,
+while $1-\sigma$ are expected to arise through sexual reproduction.
+Within the sexually-reproduced offspring,
+$S$ are born via self-fertilisation while the rest
+have parents drawn at random from the previous generation.
+Depending on the simulation software, this random drawing of parent may occur
+either with or without replacement. When drawing occurs with replacement, a small
+amount of ``residual'' selfing is expected, so that the realized selfing probability
```
Comment (molpopgen) [inline paper.tex:612]: This is the issue that we will need to hammer out in the spec.  I'm personally annoyed that I missed this in fwdpy11. Grr....

File: paper.tex [hunk 5]
```diff
@@ -476,137 +476,156 @@ \section*{Appendix}
...
+Simulation software that allow variable rates of selfing are expected to clearly
+document the expected behaviour.
+
+By allowing the definition of selfing and cloning probabilities, we allow many 
+standard models to be defined.
+However, by parameterising selfing and cloning as we have, we assume that these
+properties of populations can be specified independently from the genetics.
+In other words, mutations that cause selfing probabilities to fluctuate within
+an epoch are not considered.
+\aprcomment{
```
Comment (molpopgen) [inline paper.tex]: I think we want a section in the Discussion outlining some obvious things that people can do w/demes that are sitting "just outside" the strict specification: hard selection, etc..

File: paper.tex (hunk 2 above)
Comment (apragsdale) [reply paper.tex]: Hudson (2002) and Tajima (?) - which did you have in mind here?

File: paper.tex (hunk 4 above)
Comment (apragsdale) [reply paper.tex:612]: I wrote this section as us being agnostic to this O(1/N) difference in behavior. I can think of good arguments for and against both directions, as well as for and against us remaining agnostic.

File: paper.tex (hunk 5 above)
Comment (apragsdale) [reply paper.tex]: I'll add a comment to that point here.

File: paper.tex (hunk 3 above)
Comment (apragsdale) [reply paper.tex]: Fixed

File: paper.tex (hunk 1 above)
Comment (apragsdale) [reply paper.tex]: Changed.

File: paper.tex (hunk 2 above)
Comment (molpopgen) [reply paper.tex]: 1983 for both.

**Comment by molpopgen** [comment]
This PR looks good to me.

File: paper.tex [hunk 6]
```diff
@@ -476,137 +476,162 @@ \section*{Appendix}
 This schema, and its accompanying documentation,
 fully describe the entities in the model and their
 relationships and the required behaviour of implementations.
-Since the schema is definitive, we will not recapitulate the details
+Since the schema is definitive,
+\aprcomment{what exactly do we mean by ``definitive''?}
```
Comment (grahamgower) [inline paper.tex]: I think this is meant as a synonym for *authoritative*.

File: paper.tex [hunk 7]
```diff
@@ -476,137 +476,162 @@ \section*{Appendix}
...
 the rationale behind key design decisions.
 
-Below, we detail
-\begin{itemize}
-    \item The population genetics model in Section~\ref{sec:appendix-pop-gen-model}
-    \item The high-level model specification in Section~\ref{sec:appendix-spec}
-    \item The desired behaviour of Demes-format parsers in Section~\ref{sec:appendix-parsers}
-    \item And the intended scope of Demes in Section~\ref{sec:appendix-scope}
-\end{itemize}
+Below we detaail the population genetics model in
```
Comment (grahamgower) [inline paper.tex]: s/detaail/detail/

File: paper.tex [hunk 8]
```diff
@@ -476,137 +476,162 @@ \section*{Appendix}
...
-    \item The desired behaviour of Demes-format parsers in Section~\ref{sec:appendix-parsers}
-    \item And the intended scope of Demes in Section~\ref{sec:appendix-scope}
-\end{itemize}
+Below we detaail the population genetics model in 
+Appendix~\ref{sec:appendix-pop-gen-model},
+the high-level, human-readable model specification in
+Appendix~\ref{sec:appendix-spec}, demographic model parsers and their
+intended behaviour in Appendix~\ref{sec:appendix-parsers}, and the
+intended scope of Demes in Appendix~\ref{sec:appendix-scope}.
+The extended figures in Appendix~\ref{sec:appendix-figures} provides
```
Comment (grahamgower) [inline paper.tex]: s/provides/provide/

File: paper.tex [hunk 9]
```diff
@@ -476,137 +476,162 @@ \section*{Appendix}
...
-mean fitnesses, respectively, properly weighted according to ancestry proportions. In this framework,
-a forward-time simulation of finite populations is equivalent to multinomial sampling of allele frequencies each
-generation (\citet[][pp 29-31]{burger2000-ul}, \citet[][pp 179-181]{crowkimura1970})
-and a backwards-time (coalescent) simulation follows the approximations described in \cite[][chapter 3]{wakeley2008-hd}.
-Further, this model assumes "soft" selection \citep{christiansen1975hard}, meaning that the dynamics of population sizes
+be described by the Wright-Fisher model.
+Briefly, generations are non-overlapping (all parents
+reproduce and die simultaneously) and the expected frequency (at birth)
+of the $i^{th}$ allele currently at frequency $p_i$
+one generation in the next generation is $p_iw_i/\bar{w}$,
```
Comment (grahamgower) [inline paper.tex]: It's hard to parse this part of the sentence. Maybe the wording could be revised?

File: paper.tex (hunk 4 above)
Comment (grahamgower) [reply paper.tex:612]: I really like the text here. I think we should remain pragmatic, which means we might have to leave it ambiguous and give a strong nod towards current simulator behaviour. Our aim should be to unify how to specify simulations, not to fix discrepancies between simulators and any given mathematical model. "We want X, but reality intervenes to make implementation quite awkward, so simulators do Y"

Did a new issue get opened to discuss this?

**Review by grahamgower — APPROVED** [review]
LGTM.

File: paper.tex (hunk 4 above)
Comment (grahamgower) [reply paper.tex:612]: Maybe the discussion about O(1/n) residual selfing should continue here: https://github.com/popsim-consortium/demes-spec/issues/43

File: paper.tex (hunk 6 above)
Comment (apragsdale) [reply paper.tex]: Gotcha. Thanks.

File: paper.tex (hunk 9 above)
Comment (apragsdale) [reply paper.tex]: I reworked this a bit, and let me know if this reads more clearly.

Comment (grahamgower) [reply paper.tex]: Much clearer, thanks!

File: paper.tex [hunk 10]
```diff
@@ -476,137 +476,170 @@ \section*{Appendix}
...
-either through ancestor-descendant relationships between the populations,
-or through processes of continuous or pulse migration.
-No other properties of the populations are specified in the model:
-we are concerned only with defining the populations, their sizes, and the
+either according to their ancestor-descendant relationships
+or through processes involving migrations.
+Few other properties of the populations are specified in the model:
+we are concerned primarily with defining the populations, their sizes, and the
 movement of individuals between those populations.
+\aprcomment{previous sentence changed to allow for selfing/cloning}
```
Comment (jeromekelleher) [inline paper.tex]: Yep, LGTM. Can remove comment.

File: paper.tex [hunk 11]
```diff
@@ -476,137 +476,170 @@ \section*{Appendix}
...
+%\krtcomment{The paragraph(s) below subsume(s)/replace(s) some of what appears in later pargraphs.
+%	I'm not making an attempt yet to revise that later content--I think other authors should
+%	weigh in.}
+
+%\krtcomment{Aaron -- I have perhaps over-simplified the description of marginal allele frequency dynamics?}
 
-\krtcomment{Aaron -- I have perhaps over-simplified the description of marginal allele frequency dynamics?}
+\subsection{Population dynamics}
 
 Within a population, we implicitly assume that allele frequency dynamics can
```
Comment (jeromekelleher) [inline paper.tex]: I think we should delete "implicitly" here - this is what we assume and we should be explicit about it.

File: paper.tex [hunk 12]
```diff
@@ -476,137 +476,170 @@ \section*{Appendix}
...
-them.
-A plethora of forwards and backwards time simulators currently implement this model (REFS).
+% In the \texttt{Discussion}, we discuss alternative models compatible
+% with this framework and outline the expectations for tools implementing
+% them.
+\aprcomment{As such, this model excludes scenarios such as ``hard selection,''
+in which population sizes are dependent on a population's mean fitness, or
+stochastic fluctuations in population size, such as defining ``carrying capacities''.}
+Many forwards and backwards time simulators currently implement this model
+\citep[e.g.,][\aprcomment{and more\dots?}]{hudson2002generating,gutenkunst2009inferring,excoffier2011fastsimcoal,jouganous2017inferring,haller2019slim,thornton2019-nu}.
```
Comment (jeromekelleher) [inline paper.tex]: Should probably put msprime in here, would look weird to leave it out.

**Review by jeromekelleher — APPROVED** [review]
This is great, and I think your approach is the only feasible solution to the 1/N issue. I vote we merge and iterate.

I've created this issue on the spec to track: https://github.com/popsim-consortium/demes-spec/issues/124

File: paper.tex (hunk 4 above)
Comment (MattHartfield) [reply paper.tex:612]: I think this is the best way of clarifying the residual selfing issue. My only suggestion - which is optional - is to highlight that the 'net' selfing rate approximates to $(1-\sigma)S$ if S >> 1/N (with a reference to [Nordborg and Donnelly 1997](https://doi.org/10.1093/genetics/146.3.1185), see their Equation 3).

**Comment by apragsdale** [comment]
Merging. Thanks all for the comments! I think it's coming together.
